### PR TITLE
chore(cors): modify cors middleware order for avoiding json parsing issue with xhr

### PR DESCRIPTION
**Pull request by GuillaumeCisco**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [ ] Create automatic tests
- [ ] No automatic tests failures
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

Fix: https://github.com/ForestAdmin/forestadmin/pull/2614

### CODE REVIEW

File: templates/app/app.hbs [hunk 1]
```diff
@@ -33,6 +27,13 @@ const corsConfig = {
...
 };
 
+app.use(morgan('tiny'));
+app.use(cors(corsConfig));
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(cookieParser());
+app.use(express.static(path.join(__dirname, 'public')));
+
 app.use('/forest/authentication', cors({
```
Comment (arnaudbesnier) [inline templates/app/app.hbs:37]: Unfortunately, this route must be declared before:
`app.use(cors(corsConfig))`

**Review by arnaudbesnier — CHANGES_REQUESTED** [review]

File: templates/app/app.hbs (hunk 1 above)
Comment (GuillaumeCisco) [reply templates/app/app.hbs:37]: Thank you. I will update, the only thing we need is to have cors before body-parser :)

Comment (guillaumedeslandes) [reply templates/app/app.hbs:37]: Any hint why @arnaudbesnier ? On the server, `app.use(cookieParser());` is after `app.use(dynamicCors);`.

Comment (arnaudbesnier) [reply templates/app/app.hbs:31]: @ghusse is the best person to explain why:
`app.use('/forest/authentication', cors({...});`
is before
`app.use(cors(corsConfig));`

See the recent contribution about it:
https://github.com/ForestAdmin/lumber/pull/484

I don't get what you have in mind about your last sentence:
> On the server, app.use(cookieParser()); is after app.use(dynamicCors);

Comment (slimee) [reply templates/app/app.hbs:31]: maybe set @ghusse a reviewer of this?

Comment (ghusse) [reply templates/app/app.hbs:31]: In my PR, `app.use('/forest/authentication', cors({...});` was before `app.use(cors(corsConfig));` but after bodyparser (it was the same as before my modification).

I don't know why the order changed in the meantime.

Anyway, I don't see any problem with calling `cors` before `bodyParser`. The code inside `cors` only needs to access to headers.

**Comment by ghusse** [comment]
👍  it's OK for me

**Review by arnaudbesnier — APPROVED** [review]
The code seems nice like that.
Not sure about the commit message.
Looks like a fix to me. If so, the message should describe the potential issue for users.

**Pull request merged**